server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -218,15 +218,16 @@ Result prepareUpdateIndexRequest(ShardId shardId, UpdateRequest request, GetResu
             return new Result(update, DocWriteResponse.Result.NOOP, updatedSourceAsMap, updateSourceContentType);
         } else {
             String index = request.index();
-            final IndexRequest finalIndexRequest = new IndexRequest(index).id(request.id())
+            IndexRequest finalIndexRequest = new IndexRequest(index).id(request.id())
                 .routing(routing)
                 .source(updatedSourceAsMap, updateSourceContentType)
                 .setIfSeqNo(getResult.getSeqNo())
                 .setIfPrimaryTerm(getResult.getPrimaryTerm())
                 .waitForActiveShards(request.waitForActiveShards())
                 .timeout(request.timeout())
-                .setRefreshPolicy(request.getRefreshPolicy())
-                .setNormalisedBytesParsed(documentSizeObserver.normalisedBytesParsed());
+                .setRefreshPolicy(request.getRefreshPolicy());
+            documentSizeObserver.setNormalisedBytesParsedOn(finalIndexRequest);
```

Comment:
Nit: to be honest, I liked the old style more. But I'm not enough into the code (especially not in how this is used in serverless) to have a strong opinion.

Reply:
Same here, wondering if this is still necessary looking at the impls of `setNormalisedBytesParsedOn`.
Would below work (or are there cases where we don't want to set bytes parsed even if > 0)?
```
if(documentSizeObserver.normalisedBytesParsed() > 0) {
  finalIndexRequest.setNormalisedBytesParsed(documentSizeObserver.normalisedBytesParsed());
}
```

Reply:
Oh I see, this is because we want someone to know _not_ to set the normalized bytes, e.g. have an implementation that says `public void setNormalisedBytesParsedOn(IndexRequest indexRequest) { /* do nothing */ }` 
Not loving the inversion, but I understand why now (as I said, not enough into code to understand how/if this could be better)